### ISSUE TITLE
Table prefixes

### ISSUE THREAD

**Issue by mdlonline** [frosit/magerun-rewritetoolset#2]
The toolset does not work on Magento installs that use a table prefix like mage_

**Comment by frosit** [comment]
Good point, i will get to it soon as i'm currently refactoring everything

**Comment by selisc** [comment]
Hi @frosit 
Same issue for me too. Have you rectified the table prefix issue?

**Comment by frosit** [comment]
Hi @selisc I haven't finished the refactor. I'm also not sure if i'm going to. It was experimental. You could probably use the following as a workaround though.

It uses the simple to use Mysqlidb class for most DB operations, found at: `src/Frosit/Utils/Mysql/MysqliDb.php`. At line 34 you can set a table prefix equal to whatever is in your local.xml. That should work, haven't tested it.

**Comment by selisc** [comment]
@frosit thanks for your solution. I will try the same and will update here.

Can you please let me know the exact command to delete the url rewrites older than 30 days?

**Comment by frosit** [comment]
Use: `n98-magerun.phar rewrites:clean:older-than 30`. I usually also add `--store all` to most of the commands in here. Be sure to have a backup when using these commands.

**Comment by selisc** [comment]
Is it right ?
`n98-magerun.phar rewrites:clean:older-than 30 --store all`

**Comment by selisc** [comment]
@frosit also let me know where the file `MysqliDb.php` is exactly located in `src/Frosit/Utils/Mysql` from Magento root.

**Comment by selisc** [comment]
@frosit As per your request I just set my table prefix at line 34 at `src/Frosit/Utils/Mysql/MysqliDb.php`

After that I just executes the command `n98-magerun.phar rewrites:clean:older-than 30` but again its shows the same error
`[Zend_Db_Statement_Exception]  SQLSTATE[42S02]: Base table or view not found`

Why?

**Comment by frosit** [comment]
Hi @selisc my assumption was false, sorry for that. You should revert that change. 
There are some hard coded table names that require the prefix to be added in the code. For the older-than command, it's a simple adjustment.

You should change line 92 at `src/Frosit/Magento/Command/Rewrites/Clean/OlderThanCommand.php` and add your prefix to "core_url_rewrite". (This is for most commands)

__example__

```php
// Original line
$query .= "FROM `core_url_rewrite` WHERE `store_id` = " . $store['store_id']...

// Line including prefix
$query .= "FROM `myprefix_core_url_rewrite` WHERE `store_id` = " . $store['store_id']...
```

I just tested this and it works. Your command is right also.

**Comment by selisc** [comment]
Hi @frosit thanks for your reply. I will try this and update here. 

Kindly confirm whether I can use this command `n98-magerun.phar rewrites:clean:older-than 30 --store all` for my Live environment. 

Because I just found that the description **rewrites:clean:older-than - Remove all dupes older than x days [development]**

Why its mentioned this for **[development]**?

Thanks.
